static/js/util/profile_edit.js

```diff
@@ -49,36 +55,66 @@ export function boundSelectField(keySet, label, options) {
     errors,
     updateProfile,
   } = this.props;
-  let onChange = value => {
+  let onChange = option => {
     let clone = _.cloneDeep(profile);
-    _.set(clone, keySet, value? value.value : '');
+    let toStore;
+    if (option.value !== null && option.value !== undefined) {
+      toStore = option.value.props.value;
```

Comment:
ok, in that case do you think it would be wise to change this line to:

```js
toStore = _.get(option, ['value', 'props', 'value'])
```

for defensive-ness' sake?